exchanges/graphcache/src/store/data.ts

```diff
@@ -414,8 +404,8 @@ export const gc = () => {
 };
 
 const updateDependencies = (entityKey: string, fieldKey?: string) => {
-  if (fieldKey !== '__typename') {
-    if (entityKey !== currentData!.queryRootKey) {
+  if (fieldKey !== '__typename' && currentData) {
```

Comment:
I think there may be some cases in which this could happen, right? I'm thinking this may be so that `cacheExchange().store.resolve` still works, right? 🤔 Is there any way in which we could ensure that an operation is still available for these cases? (just very loose thoughts)

Reply:
Hmm, I think this is needed so we don't crash if someone manages to call this without an initialised data state 😅 all though that should not be possible anymore. That would mean we start marking `currentData` as always present which feels a hard one from a typing perspective as it relies on us calling `initDataState`